### DISCUSSION TITLE
Subscriptions: adding payload to ConnectionInit message

### DISCUSSION THREAD

**Discussion by fnzr** [urql-graphql/urql#2891] [Q&A] [answered]
When using GraphQL subscriptions with graphql-ws, I want to pass a user token on the payload of the ConnectionInitMessage to accept or refuse the connection.

I see I can pass arbitrary payload on the Subscribe message using the `context` key, but I couldnt find the equivalent for the ConnectionInitMessage.

Is this possible?

**Comment by JoviDeCroock** [comment, marked as answer]
You can probably pass them to `graphql-ws` as [connectionParams](https://github.com/enisdenjo/graphql-ws/blob/master/src/client.ts#L215-L226)
